protos/file.proto

```diff
@@ -19,7 +19,191 @@
 syntax = "proto3";
 
 package lance.file;
+
+// Lance v2 File Format
+// 
+// +------------------------------+
+// | Page 0, Buffer 0             |
+// | ...                          |
+// | Page 0, Buffer N             |
+// | (optional padding)           |
+// | Page 1, Buffer 0             |
+// | ...                          |
+// | Page N2, Buffer N3           |
```

Comment:
The labels `N2` and `N3` are confusing. Perhaps they should be different labels, described in text above. For example:

> `M` columns, `N[m]` pages for column `m` in `0..M`.

Reply:
I'll think this over.  I started with variables but since there were 5 of them I had `N`, `M`, `X`, `Y`, and `Z` and so I wasn't sure that was very readable either.  I wanted to avoid implying that pages have the same # of buffers or that there are the same number of metadata pages as there are column pages, etc.